requests/models.py

```diff
@@ -363,8 +364,8 @@ def prepare_url(self, url, params):
 
         # Only want to apply IDNA to the hostname
         try:
-            host = host.encode('idna').decode('utf-8')
-        except UnicodeError:
+            host = idna.encode(host).decode('utf-8')
```

Comment:
Do we want to use ``idna.encode(host, uts46=True)`` here? It sounds like that option will enable a mapping between upper and lower case characters and other "safe" mappings, but I admit I just skimmed the document. Without it any uppercase letters here will result in an ``InvalidURL``.